demo-apps/cf-secure/src/main/java/org/eclipse/californium/examples/CredentialsUtil.java

```diff
@@ -0,0 +1,203 @@
+		 */
+		X509
+	}
+
+	public static final Mode[] DEFAULT_MODES = { Mode.PSK, Mode.RPK, Mode.X509 };
+
+	// from ETSI Plugtest test spec
+	public static final String PSK_IDENTITY = "password";
+	public static final byte[] PSK_SECRET = "sesame".getBytes();
+
+	// from demo-certs
+	public static final String SERVER_NAME = "server";
+	public static final String CLIENT_NAME = "client";
+	private static final String TRUST_NAME = "root";
+	private static final char[] TRUST_STORE_PASSWORD = "rootPass".toCharArray();
+	private static final char[] KEY_STORE_PASSWORD = "endPass".toCharArray();
+	private static final String KEY_STORE_LOCATION = "certs/keyStore.jks";
+	private static final String TRUST_STORE_LOCATION = "certs/trustStore.jks";
+
+	/**
+	 * Parse arguments to modes.
+	 * 
+	 * @param args arguments
+	 * @return array of modes. if the arguments are empty,
+	 *         {@link #DEFAULT_MODES} is returned.
+	 */
+	public static Mode[] parse(String[] args) {
+		int len = args.length;
+		if (0 == len) {
+			return DEFAULT_MODES;
+		}
+		Mode[] modes = new Mode[len];
+		for (int index = 0; index < len; ++index) {
+			modes[index] = Mode.valueOf(args[index]);
+		}
+		return modes;
+	}
+
+	/**
+	 * Setup credentials for DTLS connector.
+	 * 
+	 * If PSK is provided and no PskStore is already set for the builder, a
+	 * {@link InMemoryPskStore} containing {@link #PSK_IDENTITY} assigned with
+	 * {@link #PSK_SECRET} is set. If PSK is provided with other mode(s) and
+	 * loading the certificates failed, this is just treated as warning and the
+	 * configuration is setup to use PSK only.
+	 * 
+	 * If RPK is provided, the certificates loaded for the provided alias and
+	 * this certificate is used as identity.
+	 * 
+	 * If X509 is provided, the trusts are also loaded an set additionally to
+	 * the credentials for the alias.
+	 * 
+	 * The Modes can be mixed. If RPK is before X509 in the list, RPK is set as
+	 * preferred.
+	 * 
+	 * Example:
+	 * <pre>
+	 * [PSK, RPK] setup for PSK an RPK.
+	 * [RPK, X509] setup for RPK and X509, prefer RPK
+	 * [PSK, X509, RPK] setup for PSK, RPK and X509, prefer X509
+	 * </pre>
+	 * 
+	 * @param config DTLS configuration builder. May be already initialized with
+	 *            PskStore.
+	 * @param certificateAlias alias for certificate to load as credentials.
+	 * @param modes list of supported mode. If a RPK is in the list before X509,
+	 *            the RPK is setup as preferred
+	 */
+	public static void setupCredentials(DtlsConnectorConfig.Builder config, String certificateAlias, Mode... modes) {
+
+		boolean psk = 0 <= indexOf(Mode.PSK, modes);
+		if (psk && config.getIncompleteConfig().getPskStore() == null) {
+			// Pre-shared secret keys
+			InMemoryPskStore pskStore = new InMemoryPskStore();
+			pskStore.setKey(PSK_IDENTITY, PSK_SECRET);
+			config.setPskStore(pskStore);
+		}
+		int x509 = indexOf(Mode.X509, modes);
+		int rpk = indexOf(Mode.RPK, modes);
+
+		if (0 <= x509 || 0 <= rpk) {
```

Comment:
I do not understand why you always put constant in first.
I know this is logically  the same but it seems more readable to put the variable in first. 
I mean you want check if `x509` greater or equals to `0`, not if `0` is lesser or equals than `x509` ...
Almost all the code was written like that and if I remember well Kai already ask you for that.

Reply:
That's a habit from coding C with MISRA :-).

Reply:
I have to agree with @sbernard31, I would also very much prefer if you could drop this habit of yours from the *good old days* (at least while working on Cf) ;-)